oscarlab/graphene
# Adding Github action to build Graphene Docker image

## Description of the changes 

This PR adds a continuous deployment action building the base Graphene Docker image for Azure Kubernetes Services via Github actions. The action is implemented in `.github/workflows/graphene-base-image.yaml`. It is only triggered on pushes to the master branch and only for the main Graphene repository `oscarlab/graphene` and not for any fork. Even if someone changes the second check, they cannot not overwrite the public Dockerhub images, since they're missing the login credentials to the Dockerhub account.

The action logs into Docker Hub via Github Secrets (which have to be defined in the Graphene Repo setting) while ensuring that the PW does not appear in public output logs, builds the AKS image, and pushes the resulting image to Docker Hub.

This PR also includes additional changes to test the scripts via Jenkins and removes the previously added Dockerfiles under `Tools/gsc/images`, since they're now generated during the Github action.



## How to test this PR? 

Run GSC Jenkins test (this tests the build of the base Graphene image), but does not test the actual Github action and the Docker Hub login. There is no way to test this before hand. You can find a successful workflow of my private repository here: https://github.com/vahldiek/graphene/actions/runs/280306800/workflow This action tough used a different branch and repository check. (otherwise the workflow is exactly the same as committed here)
